pandas/core/generic.py

```diff
@@ -6219,8 +6220,12 @@ def convert_dtypes(
                 for col_name, col in self.items()
             ]
             if len(results) > 0:
+                result = concat(results, axis=1, copy=False)
+                cons = cast(Type[DataFrame], self._constructor)
```

Comment:
Not super happy with this. Was getting the error
```bash
pandas/core/generic.py:6222: error: Argument 1 to "NDFrame" has incompatible type "DataFrame"; expected "Union[ArrayManager, SingleArrayManager, BlockManager, SingleBlockManager]"  [arg-type]
```
at https://github.com/pandas-dev/pandas/runs/4114882794?check_suite_focus=true, which refers to the line
`result = self._constructor(concat(results, axis=1, copy=False))`.

Bit confused because `NDFrame._constructor` raises an abstractmethod error, I'm guessing mypy is then reading the parent class `PandasObject` which returns `type(self)` i.e. in this case `NDFrame`. Then we're calling `NDFrame` with a `DataFrame` (the result of `concat` with more that 1 column). Finally the mypy type error seems to come from `DataFrame` in the `__init__` of `NDFrame` which specifies the first argument is a `Manager` and not a DataFrame. 

I guess given the return signature of `concat` is `Series | DataFrame` this is technically correct, but it seems awkward to use the `DataFrame` directly in the definition of the more general `NDFrame`